plugin/trino-hive/src/main/java/io/trino/plugin/hive/fs/CachingDirectoryLister.java

```diff
@@ -82,17 +86,25 @@ public CachingDirectoryLister(
                 .shareNothingWhenDisabled()
                 .recordStats()
                 .build();
-        this.tablePredicate = matches(tables);
+        this.tablePredicate = matches(includedTables, excludedTables);
```

Comment:
Can re-use the `matches` function definition here like `matches(includedTables).and(not(matches(excludedTables)))`

Reply:
yeah sure that is fine, happy to do that